metaflow/plugins/argo/argo_workflows.py

```diff
@@ -2066,13 +2075,16 @@ def _container_templates(self):
                         minutes_between_retries=minutes_between_retries,
                     )
                     .metadata(
-                        ObjectMeta().annotation("metaflow/step_name", node.name)
+                        ObjectMeta()
+                        .annotation("metaflow/step_name", node.name)
                         # Unfortunately, we can't set the task_id since it is generated
                         # inside the pod. However, it can be inferred from the annotation
                         # set by argo-workflows - `workflows.argoproj.io/outputs` - refer
                         # the field 'task-id' in 'parameters'
                         # .annotation("metaflow/task_id", ...)
                         .annotation("metaflow/attempt", retry_count)
+                        .annotations(resources["annotations"])
+                        .labels(resources["labels"])
```

Comment:
any reason not to set base annotations/labels here? it might be helpful if at the top of the file, we can add a comment regarding where annotations/labels apply and where they don't - would be helpful in preventing future regressions.

Reply:
these are already set on the workflow template level so they should be inherited. I'll do a quick verify on this still

Reply:
confirmed that these are inherited from the workflow template to the workflow, and workflow pods as expected.